### DISCUSSION TITLE
Use webpacker for JS dependencies

### DISCUSSION THREAD

**Discussion by grzuy** [rubygems/rubygems.org#3098] [Ideas]
I think it would be good to have `webpacker`/`webpack` for managing and updating JS depedencies more easily.

**Comment by grzuy** [comment]
Do others agree this would be a good idea?

If so I could open PR with this.

**Comment by sonalkr132** [comment]
> managing and updating JS depedencies more easily.

we don't have any js dependency beside jquery (which can probably be removed as well). No plans or need for ES6 either.
is this still useful?

**Comment by sonalkr132** [comment]
Hi, I am closing this. Feel free to re-open if you think this would be of significant help for us.

**Comment by grzuy** [comment]
Hi @sonalkr132,

Thanks for the feedback.

> > managing and updating JS depedencies more easily.
> 
> we don't have any js dependency beside jquery (which can probably be removed as well).

As far as I can see there's also 2 extra JS dependencies in `vendor/assets/javascripts`:
  - [clipboard](https://www.npmjs.com/package/clipboard), which is in v1.5.5 (4 years old version - never updated since introduction)
  - [mdo/github-buttons](https://github.com/mdo/github-buttons), unknown version

> No plans or need for ES6 either.
> is this still useful?

Apart from the usefulness of keeping the existing dependencies up to date more easily, opening this issue was also motivated by the fact that I was having trouble introducing a new JS dependency in the WebAuthn pull request (https://github.com/rubygems/rubygems.org/pull/2108#discussion_r327321092). The complication was due to the fact that `@github/webauthn-json`, isn't even considering people copy-pasting their source code in their apps as an expected installation dev flow so they don't provide dist files. So one needs to build/generate it's own dist file, which is a bit suboptimal and in some way error prone, compared to `npm install @github/webauthn-json` or any other future JS depdenency.

Not a huge thing - but wanted to give more details behind my original motivation if helpful.

Thanks again.

**Comment by sonalkr132** [comment]
> keeping the existing dependencies up to date more easily

this is valid point. I think our intention with vendoring everything is to avoid depending on third party system for build/deployment.
We should have definitely kept clipboard up to date, and manual copy paste is good enough as long as we had only one dependency to update. At this point, github-buttons is "our" code given that upstream hasn't had udpates in 5 years and we have made quite a few changes to it.
I am not sure where we stand with adding one more JS dependency and nodejs/npm. Traditionally we have been very conservative about adding dependencies to our app. 

I guess this should be open for discussion.
